pravega-sensor-collector/src/main/java/io/pravega/sensor/collector/file/FileProcessor.java

```diff
@@ -216,14 +221,31 @@ void processFile(FileNameWithOffset fileNameWithBeginOffset, long firstSequenceN
         }
     }
 
+    void moveCompletedFile(FileNameWithOffset fileNameWithBeginOffset, long endOffset, long nextSequenceNumber, Optional<UUID> txnId) throws SQLException, IOException {
+        FileUtils.moveCompletedFile(fileNameWithBeginOffset, movedFilesDirectory);
```

Comment:
Do you see that adding file to completedFiles table first followed by actually moving files make more sense.
Let us consider a scenario where we could process event from file but while moving file if there are some exception then in both place (DB table and moved folder) this file entry won't be there. Instead If we add that entry atleast in DB then next time we will not this file for processing.